InnerEye/ML/dataset/full_image_dataset.py

```diff
@@ -298,19 +299,14 @@ def get_mask_channel_or_default() -> Optional[Path]:
             return get_paths_for_channel_ids(channels=[mask_channel])[0]
 
     def get_paths_for_channel_ids(channels: List[str]) -> List[Path]:
+
         if len(set(channels)) < len(channels):
             raise ValueError(f"ids have duplicated entries: {channels}")
 
-        paths: List[Path] = []
         rows = dataframe.loc[dataframe[CSV_SUBJECT_HEADER] == patient_id]
-        for channel_id in channels:
-            row = rows.loc[rows[CSV_CHANNEL_HEADER] == channel_id]
-            if len(row) == 0:
-                raise ValueError(f"Patient {patient_id} does not have channel '{channel_id}'")
-            elif len(row) > 1:
-                raise ValueError(f"Patient {patient_id} has more than one entry for channel '{channel_id}'")
-            image_path = local_dataset_root_folder / row[CSV_PATH_HEADER].values[0]
-            paths.append(image_path)
+
+        paths: List[Path] = validated_channel_ids(channels, rows, local_dataset_root_folder, patient_id)
```

Comment:
the name of the function is somewhat misleading. it does not give validated channel ids, it converts from channel IDs to file paths.

Reply:
1) Changed function name from validated_channel_ids to converts_channels_to_file_paths

2) Function converts_channels_to_file_paths now implemented in full_image_dataset.py instead of full_image_dataset.py since line 21 from  from full_image_dataset.py (InnerEye.ML.dataset.full_image_dataset import load_dataset_sources) was not being imported in Ubuntu 20.04 -- error https://github.com/microsoft/InnerEye-DeepLearning/runs/2453669197?check_suite_focus=true 

This was due to recursive reference from import modules from full_image_dataset.py and InnerEye.ML.dataset.full.py.